### DISCUSSION TITLE
[feature request] ability to cancel a previous request when using createResource

### DISCUSSION THREAD

**Discussion by FrenchDilettante** [solidjs/solid#2129] [Ideas]
Hi folks!

I'm working on a hobby project and decided to use SolidJS. It's been a very interesting journey so far, so before I dive into my feature request I wanted to thank you for the work put into this framework.

Consider a (simplified) search feature:
```jsx
export default function Search() {
  const [inputValue, setInputValue] = createSignal("");
  const [results] = createResource(
    inputValue,
    async (q) => {
      const response = await fetch(`/api/search?q=${encodeURIComponent(q)}`);
      return await response.json();
    },
  );

  return (
    <>
      <input
        type={"search"}
        value={inputValue()}
        onKeyUp={(e) => setInputValue(e.currentTarget.value)}
      />
      <Show when={!!results.latest}>
        <ul>
          <For each={results.latest!}>{(result) => <li>{result}</li>}</For>
        </ul>
      </Show>
    </>
  );
}
```

Because this search is triggered on each keystroke, I would like the ability to cancel any ongoing request. I know that `createResource` draws inspiration from Tanstack Query, so something like:

```js
const [results] = createResource(
  inputValue,
  async (q, { signal }) => {
    const response = await fetch(`/api/search?q=${encodeURIComponent(q)}`, { signal });
    return await response.json();
  }, 
);
```

Where signal would automatically be called on each subsequent request.

If you think this is a worthwhile improvement, I'd be delighted to help implement it!

**Comment by atk** [comment]
I implemented this a few years ago and it was not accepted, since it is only required for some use cases. Use [makeAbortable from solid-primitives](https://github.com/solidjs-community/solid-primitives/blob/main/packages/resource/README.md#makeabortable) instead.
